sklearn/impute/_base.py

```diff
@@ -240,20 +240,29 @@ def fit(self, X, y=None):
                                  "== 0 and input is sparse. Provide a dense "
                                  "array instead.")
             else:
-                self.statistics_ = self._sparse_fit(X,
+                self.statistics_, _missing_mask = self._sparse_fit(X,
                                                     self.strategy,
                                                     self.missing_values,
                                                     fill_value)
+
+                sparse_constructor = (sparse.csr_matrix if X.format == 'csr'
+                                      else sparse.csc_matrix)
+                self._missing_mask = sparse_constructor(
+                    (_missing_mask, X.indices.copy(), X.indptr.copy()),
+                    shape=X.shape, dtype=bool)
+
         else:
-            self.statistics_ = self._dense_fit(X,
+            self.statistics_, self._missing_mask = self._dense_fit(X,
```

Comment:
Yup, that was something I was a aware of, I just needed a confirmation for the approach though. And yes, I am continuing the work with would not require to store the mask.

Thanks a lot.

Reply:
One thing though, I will need the missing mask during the `transform` of the `Imputer`, as we fit on the missing mask. If the `mask` 's feature has no missing values then in the transform of `SimpleImputer` `valid_statistics_indexes` chops off that feature making that mask not usable to transform of the indicator class. It would become a little complex (in terms of code) to create missing mask in the transform function of `Imputer`.

It already sound complex.

Thanks.